### PR TITLE
Feature/add rtc

### PR DESCRIPTION
**Pull request by Warukira**
## Description

1. New feature: Add the DS1307 RealTime Clock.
2. Re-enable building for all environments/ languages.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Screenshots
Screenshot 2020-05-19 at 15 23 02

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: airrohr-firmware/airrohr-firmware.ino [hunk 1]
```diff
@@ -3457,6 +3468,48 @@ void fetchSensorSPH0645(String& s){
...
+		// rtc.adjust(DateTime(F(__DATE__), F(__TIME__)));
+		// Set explicit time for example 19th March 2020 at 12 noon.
+		// rtc.adjust(DateTime(2020, 5, 19, 12, 00, 00));
+	}
+}
+
+/*****************************************************************
+ * read RTC sensor values                                        *
+*****************************************************************/
+static void fetchSensorRTC(String &s) {
```
Comment (chegejames) [inline airrohr-firmware/airrohr-firmware.ino]: @Warukira why do we need a function that only prints out date values?

File: airrohr-firmware/airrohr-firmware.ino [hunk 2]
```diff
@@ -3457,6 +3468,48 @@ void fetchSensorSPH0645(String& s){
...
+	debug_outln_verbose(FPSTR(DBG_TXT_START_READING), "RTC");
+
+	DateTime now = rtc.now();
+
+	if(send_now) {
```
Comment (chegejames) [inline airrohr-firmware/airrohr-firmware.ino]: @Warukira this seems to be a repeat of what's happening in a previous function. Good for debugging but not useful.

File: airrohr-firmware/airrohr-firmware.ino [hunk 3]
```diff
@@ -4508,6 +4589,12 @@ void loop(void) {
       		sum_send_time += sendCFA(result_PPD, PPD_API_PIN, FPSTR(SENSORS_PPD42NS), "PPD_");
 			sum_send_time += sendSensorCommunity(result_PPD, PPD_API_PIN, FPSTR(SENSORS_PPD42NS), "PPD_");
 		}
+		if (cfg::rtc_read) {
```
Comment (chegejames) [inline airrohr-firmware/airrohr-firmware.ino]: @Warukira we don't need to send RTC values as they are only timestamps without any sensor data.

File: airrohr-firmware/airrohr-firmware.ino [hunk 4]
```diff
@@ -4450,6 +4512,25 @@ void loop(void) {
...
+		Serial.print('/');
+		Serial.print(now.day(), DEC);
+		Serial.print(' ');
+		Serial.print(now.hour(), DEC);
+		Serial.print(':');
+		Serial.print(now.minute(), DEC);
+		Serial.print(':');
+		Serial.print(now.second(), DEC);
+		Serial.println();
+		fetchSensorRTC(result_RTC);
```
Comment (chegejames) [inline airrohr-firmware/airrohr-firmware.ino]: @Warukira looking at the function result_RTC does not seem to be updated anywhere.

**Review by chegejames — CHANGES_REQUESTED** [review]

File: airrohr-firmware/airrohr-firmware.ino (hunk 1 above)
Comment (Warukira) [reply airrohr-firmware/airrohr-firmware.ino]: @chegejames , we could add more functionality. This one gets the current time and prints it out.

File: airrohr-firmware/airrohr-firmware.ino (hunk 3 above)
Comment (Warukira) [reply airrohr-firmware/airrohr-firmware.ino]: Noted @chegejames

File: airrohr-firmware/airrohr-firmware.ino (hunk 1 above)
Comment (chegejames) [reply airrohr-firmware/airrohr-firmware.ino]: @Warukira we want to have a timestamp with each sensor reading.

File: airrohr-firmware/airrohr-firmware.ino (hunk 4 above)
Comment (Warukira) [reply airrohr-firmware/airrohr-firmware.ino]: I've resolved this.

File: airrohr-firmware/airrohr-firmware.ino (hunk 2 above)
Comment (Warukira) [reply airrohr-firmware/airrohr-firmware.ino]: @chegejames , was using it for debugging, but sorting it out.

**Pull request merged**